src/openforms/registrations/contrib/objects_api/api/views.py

```diff
@@ -49,3 +60,50 @@ class ObjecttypeVersionsListView(ListMixin, views.APIView):
     def get_objects(self) -> list[dict[str, Any]]:
         with get_objecttypes_client() as client:
             return client.list_objecttype_versions(self.kwargs["submission_uuid"])
+
+
+class TargetPathsListView(views.APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+
+    @extend_schema(
+        request=TargetPathsInputSerializer, responses={200: TargetPathsSerializer}
+    )
+    def post(self, request: Request, *args: Any, **kwargs: Any):
+        input_serializer = TargetPathsInputSerializer(data=request.data)
+        input_serializer.is_valid(raise_exception=True)
+
+        # Regex taken from django.urls.converters.UUIDConverter
+        match = re.search(
+            r"([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\/?$",
+            input_serializer.validated_data["objecttype_url"],
+        )
+        if not match:
+            raise exceptions.ParseError(detail={"objecttype_url": _("Invalid URL.")})
+
+        objecttype_uuid = match.group()
+
+        with get_objecttypes_client() as client:
+
+            json_schema = client.get_objecttype_version(
+                objecttype_uuid, input_serializer.validated_data["objecttype_version"]
+            )["jsonSchema"]
+
+        return_data = [
+            {
+                "target_path": json_path.segments,
+                "is_required": json_path.required,
+                "json_schema": json_schema,
+            }
+            for json_path, json_schema in iter_json_schema_paths(
+                json_schema, fail_fast=False
+            )
+            if not isinstance(json_schema, InvalidReference)
+            if json_schema_matches(
+                input_serializer.validated_data["variable_json_schema"], json_schema
+            )
+        ]
+
+        output_serializer = TargetPathsSerializer(many=True, data=return_data)
+        output_serializer.is_valid(raise_exception=True)
+        return Response(data=output_serializer.validated_data)
```

Comment:
Probably best to reraise a 500 instead here? A validation error on the output serializer means something is wrong on our end, not from the API consumer.

Reply:
discussedin office, should use `instance=` kwarg instead of `data` and then you can output with `serializer.data` which doesn't require `is_valid` to be called (that doesn't make sense in output serializers)